src/extensions/Relude_Extensions_BoundedEnum.re

```diff
@@ -27,4 +32,240 @@ module BoundedEnumExtensions = (E: Relude_Interface.BOUNDED_ENUM) => {
       };
       go([], start, startInt);
     };
+
+  /**
+   [BoundedEnumExtensions.inverseMapEqBy] takes an equality function of type
+   [(a, a) => bool] along with a function of type [t => a], where each [a]
+   is a unique value, and returns a new function of type [a => option(t)].
+
+   [inverseMapEqBy] performs the same operations as [inverseMapEq], but
+   it accepts an equality function for [a], instead of a first-class
+   [Eq] module.
+
+   Note: since there is no way to determine the ordering of [a] without
+   a comparator function, any call to the returned function must iterate
+   through the entire list of possible values until it finds a match or
+   reaches the end of the list. Therefore, the worst-case time complexity
+   of the returned function is **linear**, and failed lookups will always
+   be worst-case.
+
+   See {!val:inverseMapOrd} or {!val: inverseMapOrdBy} for a similar function
+   that accepts a comparator for [a], and returns a reverse-lookup function
+   with a worst-case running time of O(log(n)).
+
+   Running time for staging: O(n)
+   Running time of the returned lookup function: O(n)
+   */
+  let inverseMapEqBy: type a. ((a, a) => bool, E.t => a, a) => option(E.t) =
+    (eqA, eToA) => {
+      let rec loop = (lst, maybeE, getNextE) => {
+        switch (maybeE) {
+        | None =>
+          // generate the closure to pass back to the caller:
+          let arr = Belt.List.toArray(lst);
+          let len = Belt.Array.length(arr);
+          let result = a => {
+            let rec find = (arr, i, limit) =>
+              if (i < limit) {
+                let (e__, a__) = Belt.Array.getUnsafe(arr, i);
+                if (eqA(a__, a)) {
+                  Some(e__);
+                } else {
+                  find(arr, succ(i), limit);
+                };
+              } else {
+                None;
+              };
+            find(arr, 0, len);
+          };
+          result;
+        | Some(e) =>
+          // Continue building the associative list:
+          loop([(e, eToA(e)), ...lst], getNextE(e), getNextE)
+        };
+      };
+
+      // Build associative list and pass a lookup function to the caller:
+      loop([], Some(E.bottom), E.succ);
+    };
+
+  let inverseMapEqBy2: type a. ((a, a) => bool, E.t => a, a) => option(E.t) =
+    (eqA, eToA) => {
+      // create an associative array for reverse lookups:
+      let arr =
+        upFromIncludingAsList(E.bottom)
+        |> Belt.List.mapU(_, (. e) => (e, eToA(e)))
+        |> Belt.List.toArray;
+      let len = Belt.Array.length(arr);
+      // generate the closure to pass back to the caller:
+      let result = a => {
+        let rec find = (arr, i, limit) =>
+          if (i < limit) {
+            let (e__, a__) = Belt.Array.getUnsafe(arr, i);
+            if (eqA(a__, a)) {
+              Some(e__);
+            } else {
+              find(arr, succ(i), limit);
+            };
+          } else {
+            None;
+          };
```

Comment:
Maybe I missed this before, but isn't this just the normal `Array` `find` function? I'm not sure, but I suspect this already exists on `Array` (of if not, it could be added), or maybe on `Belt.Array`. Does it need to be defined inline like this?

Reply:
Yeah, I think you're right. I'll revisit that.